api/src/main/java/org/pmiops/workbench/monitoring/GaugeRecorderService.java

```diff
@@ -0,0 +1,44 @@
+package org.pmiops.workbench.monitoring;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.pmiops.workbench.monitoring.views.OpenCensusStatsViewInfo;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GaugeRecorderService {
+  private static final Logger logger = Logger.getLogger(GaugeRecorderService.class.getName());
+
+  private final List<GaugeDataCollector> gaugeDataCollectors;
+  private final MonitoringService monitoringService;
+
+  public GaugeRecorderService(
+      List<GaugeDataCollector> gaugeDataCollectors, MonitoringService monitoringService) {
+    this.gaugeDataCollectors = gaugeDataCollectors;
+    this.monitoringService = monitoringService;
+  }
+
+  public Map<OpenCensusStatsViewInfo, Number> record() {
+    ImmutableMap<OpenCensusStatsViewInfo, Number> result =
+        gaugeDataCollectors.stream()
+            .map(GaugeDataCollector::getGaugeData)
+            .flatMap(m -> m.entrySet().stream())
+            .collect(ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
I'm not sure what happens with the immutable map collector if you have duplicate keys.

Either way, I would document what the expected behavior here is (probably on GaugeDataCollector) and probably add a test for it. I think ideally it should be one of the following:
- No two GaugeDataCollectors may export the same data key, or
- Values are grouped by key and summed across all GaugeDataCollectors